### DISCUSSION TITLE
Published PKI maturity assessment tool needs to be fixed !

### DISCUSSION THREAD

**Discussion by joaovazmartins** [pkic/community#31] [PKI Maturity Model (PKIMM)]
HI community,

After downloading the **PKI maturity assessment tool** published in the URL:

https://docs.google.com/spreadsheets/d/1C1CARkzTRvImmH5FtMx6_Yu56PASGVSY/export?usp=drive_link&ouid=102943605657406341516&rtpof=true&sd=true

Going through the Excel file mentioned above I noticed a small error in the "Assessment" sheet (G41) which was showing NULL content.
By checking the formula that took me to the "source" sheet, where i could see that in (U85) was missing "#### Guidance" (which should be two lines below, U87).

Tried to follow the input instructions (A4:A13), which tell us to copy/paste from the Github repository but that ended up in several other errors.

Although I have some knowledge of Excel, That is not my main core experience.
Can someone help fix the error and publish a corrected file on the PKI Consortium web page? 

Best Regards,
Joao Vaz Martins

**Comment by Jirka3Key** [comment]
Hello,

you correctly identified the missing heading (the string "#### Guidance") was the reason why cell G41 on the Assessment sheet was blank.

To verify that, I did the following:
I downloaded the Tool from your link
I added again the content to column U on the "Source" sheet according to the instructions (lower box only)
Cell G4 on the "Assessment" sheet is already filled in correctly
I did not interfere with Tool's formulas

The explanation may therefore be that the Guidance heading for the requirement "List of relevant laws, regulations, and standards, exist and is maintained" was missing.

Best Regards,
Jirka Lucky

**Reply by joaovazmartins** [reply]
As I mentioned above, I tried to follow the input instructions (A11:A13), which tell us to copy/paste from the Github repository but that ended up in several other errors (even after replacing the "=" with "").

Would you please share the corrected Excel file on the PKI Consortium webpage (Link below)?

https://docs.google.com/spreadsheets/d/1C1CARkzTRvImmH5FtMx6_Yu56PASGVSY/export?usp=drive_link&ouid=102943605657406341516&rtpof=true&sd=true

**Reply by joaovazmartins** [reply]
I believe you wanted to say 

"Cell **G41** on the "Assessment" sheet is already filled in correctly" instead of **G4**

But the PKI Consortium webpage still has the **uncorrected** Excel file.

**Reply by joaovazmartins** [reply]
another recommendation

the Excel File uses some new Functions only available in the most recent release of Excel Office 365.
Functions like TEXTBEFORE() and TEXTAFTER() are not available even in **MS Excel 2021** (which shows "**_xlfn.**" before those functions to identify incompatible functions).
   
See for example cell **W69** in the "source" sheet, what I see is the below content
=@IF(AND(LEFT(U69,1)="#",IFERROR(LEN(@_**xlfn**.TEXTBEFORE(U69," "))=4,0)),_**xlfn**.TEXTAFTER(U69," "),W68)

better not to use such new functions and instead use alternative ones to avoid any inconvenience for PKI Consortium members in case they want to follow GITHUB update instructions.

**Comment by 3keyroman** [comment]
Hello @joaovazmartins ,

we've recently rolled out an update to our assessment tool, which is now available for your use at https://pkic.org/pkimm/tools/ (`PKI_Maturity_Assessment_Tool_v240318`).

It's important to note that at present, the tool is optimized for use with Excel for Microsoft 365 or Excel for web due to their formula support. However, rest assured that we are actively considering making it backward compatible in future updates to ensure accessibility across various platforms.

Your feedback and suggestions are invaluable to us, so please don't hesitate to share any thoughts or updates you may have regarding the assessment tool.

Thank you!

With best regards,
Roman

**Reply by joaovazmartins** [reply]
Hi Roman,

   Apologies for the late reply (due to Easter Holidays).

   Noted that the current release of the assessment tool published on the website (URL: **https://pkic.org/pkimm/tools/**  is optimized for use with Excel for Microsoft 365 or Excel for web.
   Also glad to know that the Consortium has understood that some members or even other organizations may not be able to use Cloud services due to their own security policies therefore Office 365 or Google web services **are not an inclusive option**.

   I'm not fully certain but I believe Microsoft is planning to release Office 2024 without Cloud licensing requirements hopefully soon (until the end of the year). Hopefully then all Excel Office 365 new functionalities will become available as well. 

Regards,
Joao Vaz Martins

**Reply by tcasett** [reply]
As a technical person interested in this tool I find it disappointing that a proprietary/closed system seems to be required.

PKI is a free and open idea https://datatracker.ietf.org/doc/html/rfc5280

Should this consortium not be publishing evaluation tools that are equally accessible?

**Comment by 3keyroman** [comment]
> As a technical person interested in this tool I find it disappointing that a proprietary/closed system seems to be required.
> 
> PKI is a free and open idea https://datatracker.ietf.org/doc/html/rfc5280
> 
> Should this consortium not be publishing evaluation tools that are equally accessible?

Thank you, @tbielawatt, for sharing your thoughts!

I appreciate your advocacy for open systems. It's essential to embrace principles like those laid out in the PKI concept, as exemplified in RFC 5280.

I want to reassure you that accessibility and openness are central to our mission. The PKI Maturity Model, along with its assessment tools, is accessible to everyone. We're actively working on enhancements, such as transitioning the self-assessment tools into web-based applications for easier access and exploring compatibility with various platforms, including open-source alternatives.

While there are some functionalities tied to Microsoft 365 in our current Excel-based tools, we're committed to ensuring compatibility across different platforms. Our resources are finite, but we're dedicated to maximizing their impact in improving the PKI Maturity Model.

Your input is incredibly valuable, and we welcome contributions from individuals like you to further enhance the accessibility and openness of the PKI Maturity Model.


With best regards,
Roman

**Reply by tcasett** [reply]
Thanks so much for the response. I appreciate hearing that feedback from you. I'm looking forward to seeing where this goes.

**Comment by vanbroup** [comment]
What do you think about the new online version of the PKI Maturity Model Self-Assessment?

https://pkic.org/pkimm/tools/self-assessment/

**Comment by csmarin** [comment]
1. Would it be possible to include Microsoft on the list, because they have recently announced full support for PQ in their new release of ADCS:
https://techcommunity.microsoft.com/blog/microsoft-security-blog/post-quantum-cryptography-comes-to-windows-insiders-and-linux/4413803
+ https://www.pkisolutions.com/webinars/adcs-changes-2025/
2. Also NIST approved the 4th PQ algorithm, HQC (Hamming Quasi-Cyclic) Currently, there is no FIPS standard assigned to the HQC (Hamming Quasi-Cyclic) algorithm. NIST has selected HQC as a post-quantum encryption candidate but has not yet assigned a FIPS number for it. The expected draft standard for HQC is anticipated in early 2026, and it is expected to be defined under FIPS 207.

**Reply by primetomas** [reply]
Which list are you referring to? This is a thread for the PKI Maturity model, it doesn't have a list really. It sounds like you are thinking about https://github.com/pkic/pqccm/

**Comment by csmarin** [comment]
Yes, you are right, it is in the context of PQC Capabilities Matrix
(PQCCM), sorry for the confusion.
